### PR TITLE
Feature/21 add adts

### LINKED ISSUE #21: Add support for ADTs

### PR DESCRIPTION
**Pull request by kevinwallimann**
Closes #21

### CODE REVIEW

File: library/src/test/scala/za/co/absa/springdocopenapiscala/OpenAPIModelRegistrationSpec.scala
Comment (jakipatryk) [inline library/src/test/scala/za/co/absa/springdocopenapiscala/OpenAPIModelRegistrationSpec.scala:1]: Let's try to add `case object` too. Not sure if this will be working out of the box, maybe something will have to be added to the code.

Comment (kevinwallimann) [reply library/src/test/scala/za/co/absa/springdocopenapiscala/OpenAPIModelRegistrationSpec.scala:1]: I added a test case. It does work out of the box. The case object is handled by the `handleCaseClass` method

Comment (jakipatryk) [reply library/src/test/scala/za/co/absa/springdocopenapiscala/OpenAPIModelRegistrationSpec.scala:1]: Ok this is great, though a bit unexpected :D Do you think we dhould rename this method then, maybe `handleCaseType`?

Comment (kevinwallimann) [reply library/src/test/scala/za/co/absa/springdocopenapiscala/OpenAPIModelRegistrationSpec.scala:1]: Sure why not, I renamed it

**Review by jakipatryk — COMMENTED** [review]
Code LGTM, I'll approve once the PR is "repointed" after #25 merge.

**Review by jakipatryk — APPROVED** [review]

**Pull request merged**
